pkg/expr/reader_test.go

```diff
@@ -0,0 +1,144 @@
+package expr
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data/utils/jsoniter"
+	data "github.com/grafana/grafana-plugin-sdk-go/experimental/apis/data/v0alpha1"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReaderReduceMode(t *testing.T) {
```

Comment:
i can only test that the expression-parsing does not return an error, there is no easy way to verify that the expression that was built, contains the correct `mode`.

Reply:
> i can only test that the expression-parsing does not return an error, there is no easy way to verify that the expression that was built, contains the correct `mode`.

Is that because the `ExpressionQuery` instance returned by `reader.ReadQuery(q, iter)` has a public `.Properties` method (excuse my Ruby OOP terminology), but that the `.data` on Properties is private, so we can't reach in and check the results?

Is it worth adding a code-comment along these lines? I tried to come up with a new name for this test, to reflect that we're only checking the errors we get back, but had a hard time coming up with a good name for this.

Reply:
actually i was wrong, and we can get access to the needed attribute, see https://github.com/grafana/grafana/pull/94837